day2/src/main.rs

```diff
@@ -0,0 +1,127 @@
+use std::env;
+use std::fmt;
+use std::{
+    fs::File,
+    io::{BufRead, BufReader},
+};
+
+pub enum ExitCodes {
+    Success,
+    GracefulExit,
+    MemOutOfBounds,
+    OpOutBounds,
+    InvalidOp,
+}
+
+impl fmt::Display for ExitCodes {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        match self {
+            ExitCodes::Success => write!(f, "Success"),
+            ExitCodes::GracefulExit => write!(f, "Graceful Exit"),
+            ExitCodes::MemOutOfBounds => write!(f, "Memory Read Out of Bounds"),
+            ExitCodes::OpOutBounds => write!(f, "Opcode Out of Bounds"),
+            ExitCodes::InvalidOp => write!(f, "Operation Not Implemented"),
+        }
+    }
+}
+
+pub struct GravityAssistParser {
+    pub ip: usize,
+    pub instructions: Vec<i32>,
+}
+
+impl GravityAssistParser {
+    pub fn update_mem(&mut self, z: usize, v: i32) -> ExitCodes {
+        if z > self.instructions.len() {
+            return ExitCodes::MemOutOfBounds;
+        }
+        self.instructions[z] = v;
+        ExitCodes::Success
+    }
+}
+
+pub fn compute(gap: &mut GravityAssistParser, op: i32, x: i32, y: i32, z: usize) -> ExitCodes {
+    let a = (&gap.instructions).get(x as usize);
+    let b = (&gap.instructions).get(y as usize);
+    match (op, a, b) {
+        (_, None, _) => ExitCodes::MemOutOfBounds,
+        (_, _, None) => ExitCodes::MemOutOfBounds,
+        (1, Some(m), Some(n)) => {
+            let s = (*m) + (*n);
+            gap.update_mem(z, s)
+        }
+        (2, Some(m), Some(n)) => {
+            let p = (*m) * (*n);
+            gap.update_mem(z, p)
+        }
+        _ => ExitCodes::InvalidOp,
+    }
```

Comment:
Ooh this is a really nice way to handle the instructions, mine was a lot less clean :sweat_smile: